### ISSUE TITLE
2.13.1: build time warnings

### ISSUE THREAD

**Issue by kloczek** [mm2/Little-CMS#303]
I'm usingg gcc 12.0.1.
Summary stats:
```console
[tkloczko@devel-g2v SPECS]$ rpmbuild -ba lcms2.spec --quiet 2>&1 | grep -- -W | sed 's/.*\[//; s/\]//' | sort | uniq -c | sort -nr
      6 -Wstringop-overflow=
      3 -Wunused-variable
      3 -Wunused-function
      3 -Wstringop-overread
      2 -Wstrict-aliasing
      1 -Wunused-result
      1 -Wunused-but-set-variable
      1 -Wmissing-braces
      1 -Wmisleading-indentation
```
And extracted warnings from stderr
```console
cmsplugin.c: In function '_cmsReadFloat32Number':
cmsplugin.c:181:15: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  181 |         *n = *(cmsFloat32Number*)(void*)&tmp;
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cmsplugin.c: In function '_cmsWriteFloat32Number':
cmsplugin.c:312:12: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  312 |     tmp = *(cmsUInt32Number*) (void*) &n;
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
jpgicc.c: In function 'TransformImage':
jpgicc.c:945:13: warning: 'cmsSetAlarmCodes' reading 32 bytes from a region of size 8 [-Wstringop-overread]
  945 |             cmsSetAlarmCodes(Alarm);
      |             ^~~~~~~~~~~~~~~~~~~~~~~
jpgicc.c:945:13: note: referencing argument 1 of type 'const cmsUInt16Number[16]'
In file included from ../../utils/common/utils.h:41,
                 from jpgicc.c:28:
../../include/lcms2.h:1789:35: note: in a call to function 'cmsSetAlarmCodes'
 1789 | CMSAPI void             CMSEXPORT cmsSetAlarmCodes(const cmsUInt16Number NewAlarm[cmsMAXCHANNELS]);
      |                                   ^~~~~~~~~~~~~~~~
In function 'TransformImage',
    inlined from 'main' at jpgicc.c:1268:5:
jpgicc.c:945:13: warning: 'cmsSetAlarmCodes' reading 32 bytes from a region of size 8 [-Wstringop-overread]
  945 |             cmsSetAlarmCodes(Alarm);
      |             ^
jpgicc.c:945:13: note: referencing argument 1 of type 'const cmsUInt16Number[16]'
../../include/lcms2.h: In function 'main':
../../include/lcms2.h:1789:35: note: in a call to function 'cmsSetAlarmCodes'
 1789 | CMSAPI void             CMSEXPORT cmsSetAlarmCodes(const cmsUInt16Number NewAlarm[cmsMAXCHANNELS]);
      |                                   ^
In function 'TransformImage',
    inlined from 'main' at jpgicc.c:1268:5:
jpgicc.c:945:13: warning: 'cmsSetAlarmCodes' reading 32 bytes from a region of size 8 [-Wstringop-overread]
  945 |             cmsSetAlarmCodes(Alarm);
      |             ^
jpgicc.c:945:13: note: referencing argument 1 of type 'const cmsUInt16Number[16]'
../../include/lcms2.h: In function 'main':
../../include/lcms2.h:1789:35: note: in a call to function 'cmsSetAlarmCodes'
 1789 | CMSAPI void             CMSEXPORT cmsSetAlarmCodes(const cmsUInt16Number NewAlarm[cmsMAXCHANNELS]);
      |                                   ^
testplugin.c: In function 'CheckMPEPlugin':
testplugin.c:982:21: warning: unused variable 'myTag' [-Wunused-variable]
  982 |     cmsUInt32Number myTag = 1234;
      |                     ^~~~~
testplugin.c: At top level:
testplugin.c:1337:55: warning: missing braces around initializer [-Wmissing-braces]
 1337 | static cmsPluginTransform FullTransformPluginSample = {
      |                                                       ^
......
 1341 |      TransformFactory
      |      {
 1342 | };
      | }
testcms2.c: In function 'ReverseSampler':
testcms2.c:509:5: warning: this 'else' clause does not guard... [-Wmisleading-indentation]
  509 |     else
      |     ^~~~
testcms2.c:521:9: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'else'
  521 |         cmsDoTransform(p ->sRGB2Lab, rgb, Out, 1);
      |         ^~~~~~~~~~~~~~
testcms2.c: In function 'CheckMD5':
testcms2.c:7355:21: warning: variable 'h' set but not used [-Wunused-but-set-variable]
 7355 |     _cmsICCPROFILE* h;
      |                     ^
testcms2.c: In function 'main':
testcms2.c:9163:20: warning: unused variable 'DoZooTests' [-Wunused-variable]
 9163 |     cmsInt32Number DoZooTests = 0;
      |                    ^~~~~~~~~~
In file included from testcms2.c:27:
testcms2.h:47:22: warning: ignoring return value of 'chdir' declared with attribute 'warn_unused_result' [-Wunused-result]
   47 | #    define CHDIR(a) chdir(a)
      |                      ^~~~~~~~
testcms2.c:9184:13: note: in expansion of macro 'CHDIR'
 9184 |             CHDIR(argv[2]);
      |             ^~~~~
testcms2.c: At top level:
testcms2.c:6517:16: warning: 'CheckCMYKRelCol' defined but not used [-Wunused-function]
 6517 | cmsInt32Number CheckCMYKRelCol(void)
      |                ^~~~~~~~~~~~~~~
testcms2.c:5171:16: warning: 'CheckDictionary16' defined but not used [-Wunused-function]
 5171 | cmsInt32Number CheckDictionary16(cmsInt32Number Pass,  cmsHPROFILE hProfile)
      |                ^~~~~~~~~~~~~~~~~
testcms2.c:2618:15: warning: 'GammaTableLinear' defined but not used [-Wunused-function]
 2618 | cmsToneCurve* GammaTableLinear(cmsInt32Number nEntries, cmsBool Dir)
      |               ^~~~~~~~~~~~~~~~
testcms2.c:832:25: warning: 'AllowedErr' defined but not used [-Wunused-variable]
  832 | static cmsFloat64Number AllowedErr = FIXED_PRECISION_15_16;
      |                         ^~~~~~~~~~
testcms2.c: In function 'CreateNamedColorProfile':
testcms2.c:3804:5: warning: 'cmsAppendNamedColor' accessing 32 bytes in a region of size 8 [-Wstringop-overflow=]
 3804 |     cmsAppendNamedColor(colors, "Hazelnut 14-1315", PCS, Colorant);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
testcms2.c:3804:5: note: referencing argument 4 of type 'cmsUInt16Number[16]'
In file included from ../include/lcms2_plugin.h:46,
                 from ../src/lcms2_internal.h:31,
                 from testcms2.h:30:
../include/lcms2.h:1388:37: note: in a call to function 'cmsAppendNamedColor'
 1388 | CMSAPI cmsBool            CMSEXPORT cmsAppendNamedColor(cmsNAMEDCOLORLIST* v, const char* Name,
      |                                     ^~~~~~~~~~~~~~~~~~~
testcms2.c:3810:5: warning: 'cmsAppendNamedColor' accessing 32 bytes in a region of size 8 [-Wstringop-overflow=]
 3810 |     cmsAppendNamedColor(colors, "Kale 18-0107", PCS, Colorant);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
testcms2.c:3810:5: note: referencing argument 4 of type 'cmsUInt16Number[16]'
../include/lcms2.h:1388:37: note: in a call to function 'cmsAppendNamedColor'
 1388 | CMSAPI cmsBool            CMSEXPORT cmsAppendNamedColor(cmsNAMEDCOLORLIST* v, const char* Name,
      |                                     ^~~~~~~~~~~~~~~~~~~
testcms2.c: In function 'CreateNamedColorProfile':
testcms2.c:3804:5: warning: 'cmsAppendNamedColor' accessing 32 bytes in a region of size 8 [-Wstringop-overflow=]
 3804 |     cmsAppendNamedColor(colors, "Hazelnut 14-1315", PCS, Colorant);
      |     ^
testcms2.c:3804:5: note: referencing argument 4 of type 'cmsUInt16Number[16]'
../include/lcms2.h:1388:37: note: in a call to function 'cmsAppendNamedColor'
 1388 | CMSAPI cmsBool            CMSEXPORT cmsAppendNamedColor(cmsNAMEDCOLORLIST* v, const char* Name,
      |                                     ^
testcms2.c:3810:5: warning: 'cmsAppendNamedColor' accessing 32 bytes in a region of size 8 [-Wstringop-overflow=]
 3810 |     cmsAppendNamedColor(colors, "Kale 18-0107", PCS, Colorant);
      |     ^
testcms2.c:3810:5: note: referencing argument 4 of type 'cmsUInt16Number[16]'
../include/lcms2.h:1388:37: note: in a call to function 'cmsAppendNamedColor'
 1388 | CMSAPI cmsBool            CMSEXPORT cmsAppendNamedColor(cmsNAMEDCOLORLIST* v, const char* Name,
      |                                     ^
testcms2.c:3804:5: warning: 'cmsAppendNamedColor' accessing 32 bytes in a region of size 8 [-Wstringop-overflow=]
 3804 |     cmsAppendNamedColor(colors, "Hazelnut 14-1315", PCS, Colorant);
      |     ^
testcms2.c:3804:5: note: referencing argument 4 of type 'cmsUInt16Number[16]'
../include/lcms2.h:1388:37: note: in a call to function 'cmsAppendNamedColor'
 1388 | CMSAPI cmsBool            CMSEXPORT cmsAppendNamedColor(cmsNAMEDCOLORLIST* v, const char* Name,
      |                                     ^
testcms2.c:3810:5: warning: 'cmsAppendNamedColor' accessing 32 bytes in a region of size 8 [-Wstringop-overflow=]
 3810 |     cmsAppendNamedColor(colors, "Kale 18-0107", PCS, Colorant);
      |     ^
testcms2.c:3810:5: note: referencing argument 4 of type 'cmsUInt16Number[16]'
../include/lcms2.h:1388:37: note: in a call to function 'cmsAppendNamedColor'
 1388 | CMSAPI cmsBool            CMSEXPORT cmsAppendNamedColor(cmsNAMEDCOLORLIST* v, const char* Name,
      |                                     ^
```

**Comment by mm2** [comment]
Thanks @kloczek for sending me the report.

Most of those warnings are in the testbed code and in the jpegicc sample, so not part of the library. Anyway I will try to fix those that makes sense. I have checked with gcc 10 and it does not complain on some, strict aliasing for example.
There is at least one read out of bounds that is detected in jpegicc, fortunately is harmless. I will definitively fix that.

**Comment by mm2** [comment]
Using  gcc (Debian 12.2.0-14) 12.2.0 the current source code raises no warning with the given flags
[log.txt](https://github.com/mm2/Little-CMS/files/12333492/log.txt)

**Comment by kloczek** [comment]
Few days ago has been released gcc 13.2 https://gcc.gnu.org/gcc-13/
